src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -358,4 +371,23 @@ private JsonObject toJson(EncryptionKey key) {
         jo.put("expires", key.getExpires().toEpochMilli());
         return jo;
     }
-}
+
+    private boolean isWithinCutOffTime(EncryptionKey key, Instant now, boolean duringRotation) {
+        int siteId = 8;
+        Duration cutoffTime;
+        switch (siteId) {
+            case -1:  cutoffTime = masterKeyRotationCutoffTime;
+                break;
+            case -2:  cutoffTime = refreshKeyRotationCutOffTime;
+                break;
+            default: cutoffTime = siteKeyRotationCutOffTime;
+                break;
+        }
```

Comment:
can you refactor to reduce the (extra but minor) processing such as:

```
if (!(filterKeyOverCutOffTime && duringRotation)) {
    return true;
}

//rest of switch and compareTo statement
....
```

Reply:
done